tests/unit/misc/test_utilcmds.py

```diff
@@ -0,0 +1,152 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015-2016 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for qutebrowser.misc.utilcmds."""
+
+import contextlib
+import logging
+import os
+import pytest
+import signal
+import time
+
+_hunter_available = False
+try:
+    import hunter  # pylint: disable=unused-import
+    _hunter_available = True
+except ImportError:
+    pass
+
+from qutebrowser.misc import utilcmds as utilcmdsmod
+
+from qutebrowser.commands import cmdexc
+
+
+@contextlib.contextmanager
+def _trapped_segv(handler):
+    """Temporarily install given signal handler for SIGSEGV."""
+    old_handler = signal.signal(signal.SIGSEGV, handler)
+    yield
+    signal.signal(signal.SIGSEGV, old_handler)
+
+
+def test_debug_crash_exception():
+    """Verify that debug_crash crashes as intended."""
+    with pytest.raises(Exception):
+        utilcmdsmod.debug_crash(typ='exception')
+
+@pytest.mark.skipif(os.name == 'nt',
+                    reason="current CPython/win can't recover from SIGSEGV")
+def test_debug_crash_segfault():
+    """Verify that debug_crash crashes as intended."""
+    caught = False
+
+    def _handler(num, frame):
+        """Temporary handler for segfault."""
+        nonlocal caught
+        caught = num == signal.SIGSEGV
+
+    with _trapped_segv(_handler):
+        # since we handle the segfault, execution will continue and run into
+        # the "Segfault failed (wat.)" Exception
+        with pytest.raises(Exception) as excinfo:
+            utilcmdsmod.debug_crash(typ='segfault')
+        time.sleep(0.001)
+    assert caught
+    assert 'Segfault failed' in str(excinfo.value)
+
+
+@pytest.mark.skipif(not _hunter_available, reason="hunter not available")
```

Comment:
You can simply do `pytest.importorskip('hunter')` inside the test instead ([docs](http://docs.pytest.org/en/latest/skipping.html?highlight=importorskip#skipping-on-a-missing-import-dependency))